packages/flutter/test/widgets/web_image_test.dart

```diff
@@ -0,0 +1,1308 @@
+    expect(provider.loadCallCount, 1);
+  });
+
+  testWidgets('evict an image during precache', (WidgetTester tester) async {
+    // This test checks that the live image tracking does not hold on to a
+    // pending image that will never complete because it has been evicted from
+    // the cache.
+    // The scenario may arise in a test harness that is trying to load real
+    // images using `tester.runAsync()`, and wants to make sure that widgets
+    // under test have not also tried to resolve the image in a FakeAsync zone.
+    // The image loaded in the FakeAsync zone will never complete, and the
+    // runAsync call wants to make sure it gets a load attempt from the correct
+    // zone.
+    final _TestWebImageProvider provider = _TestWebImageProvider('example.com/pic.jpg');
+
+    await tester.runAsync(() async {
+      final List<Future<void>> futures = <Future<void>>[];
+      await tester.pumpWidget(Builder(builder: (BuildContext context) {
+        futures.add(precacheWebImage(provider, context));
+        webImageCache.evict(provider.src);
+        futures.add(precacheWebImage(provider, context));
+        provider.complete(image10x10);
+        return const SizedBox.expand();
+      }));
+      await Future.wait<void>(futures);
+      expect(webImageCache.statusForKey(provider.src).keepAlive, true);
+      expect(webImageCache.statusForKey(provider.src).live, true);
+
+      // Schedule a frame to get precacheWebImage to stop listening.
+      SchedulerBinding.instance.scheduleFrame();
+      await tester.pump();
+      expect(webImageCache.statusForKey(provider.src).keepAlive, true);
+      expect(webImageCache.statusForKey(provider.src).live, false);
+
+      webImageCache.clear();
+    });
+  });
+
+  testWidgets('errorBuilder - fails on load', (WidgetTester tester) async {
+    final UniqueKey errorKey = UniqueKey();
+    late Object caughtException;
+    await tester.pumpWidget(
+      WebImage(
+        image: _FailingWebImageProvider('example.com/pic.jpg', failOnLoad: true, throws: 'threw', image: image10x10),
+        errorBuilder: (BuildContext context, Object error, StackTrace? stackTrace) {
+          caughtException = error;
+          return SizedBox.expand(key: errorKey);
+        },
+      ),
+    );
+
+    await tester.pump();
+
+    expect(find.byKey(errorKey), findsOneWidget);
+    expect(caughtException.toString(), 'threw');
+    expect(tester.takeException(), isNull);
+  });
+
+  testWidgets('no errorBuilder - failure reported to FlutterError', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      WebImage(
+        image: _FailingWebImageProvider('example.com/pic.jpg', failOnLoad: true, throws: 'threw', image: image10x10),
+      ),
+    );
+
+    await tester.pump();
+
+    expect(tester.takeException(), 'threw');
+  });
+}
+
+web_shim.HTMLImageElement createTestWebImage(
+    {required int width, required int height}) {
+  final web_shim.HTMLCanvasElement canvas =
```

Comment:
Would `web_shim.HTMLCanvasElement()` work here?

Reply:
No. The constructor can't be directly invoked.